test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -429,24 +535,82 @@ define([
                 });
 
                 it('can start in an internal error state', async function () {
+                    setUpCellInState(this, 'INTERNAL_ERROR');
+                    await startRunningCell(this);
+
+                    expect(this.appCellWidgetInstance.__fsm().getCurrentState().state).toEqual(
+                        fsmState.INTERNAL_ERROR
+                    );
+                    // the reset button should be active
+                    const resetButton = this.kbaseNode.querySelector(selectors.reset);
+                    expect(resetButton).not.toHaveClass('hidden');
+                    expect(resetButton).not.toHaveClass('disabled');
+                });
+
+                it('shows an internal error if the FSM throws a hissy fit', async function () {
```

Comment:
FSM errors were not displaying in a helpful way, so this rectifies the situation.